CodeGen/Generators/NanoFrameworkGenerator.cs

```diff
@@ -126,85 +122,183 @@ public static void Generate(string rootDir, Quantity[] quantities, bool updateNa
-            CancellationToken cancellationToken, bool updateNanoFrameworkDependencies, string outputDir)
+        /// <summary>
+        /// Updates existing nanoFramework projects and nuspecs with the latest versions.
+        /// </summary>
+        /// <param name="rootDir">The root directory</param>
+        /// <param name="quantities">The quantities to update nuspecs</param>
+        public static void UpdateNanoFrameworkDependencies(
+            string rootDir,
+            Quantity[] quantities)
         {
-            if (updateNanoFrameworkDependencies)
+            // working path
+            string path = Path.Combine(rootDir, "UnitsNet.NanoFramework\\GeneratedCode");
+
+            Log.Information("");
+            Log.Information("Updating .NET nanoFramework references using nuget CLI");
+
+            // run nuget CLI
+            var nugetCLI = new System.Diagnostics.Process
             {
-                logger.LogInformation("Updating nanoFramework dependencies.");
-
-                // mscorlib
-                IEnumerable<NuGetVersion> packageVersions = resource.GetAllVersionsAsync(
-                    "nanoFramework.CoreLibrary",
-                    cache,
-                    logger,
-                    cancellationToken).Result;
-
-                // get NuGet package Version for mscorlib
-                // grab latest available (doesn't matter if it's preview or stable)
-                NuGetVersion mscorlibVersion = packageVersions.OrderByDescending(v => v).First();
-                MscorlibVersion = mscorlibVersion.Version.ToString();
-                MscorlibNuGetVersion = mscorlibVersion.ToNormalizedString();
-
-                // System.Math
-                packageVersions = resource.GetAllVersionsAsync(
-                    "nanoFramework.System.Math",
-                    cache,
-                    logger,
-                    cancellationToken).Result;
-
-                // grab latest available (doesn't matter if it's preview or stable)
-                // making an assumption here that the available version is referencing the correct mscolib
-                var mathVersion = packageVersions.OrderByDescending(v => v).First();
-                MathVersion = mathVersion.Version.ToString();
-                MathNuGetVersion = mathVersion.ToNormalizedString();
-
-                logger.LogInformation($"Referencing nanoFramework.CoreLibrary {MscorlibNuGetVersion}");
-                logger.LogInformation($"Referencing nanoFramework.System.Math {MathNuGetVersion}");
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "nuget.exe",
+                    Arguments = $"update {path}\\UnitsNet.nanoFramework.sln -PreRelease",
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                }
+            };
+
+            nugetCLI.Start();
+
+            // start nuget CLI and wait for exit
+            if (!nugetCLI.Start())
```

Comment:
Aren't you calling Start twice here?

Reply:
Ooops... Yes. My bad. This poped after a final refactor and slipped my review...